### ISSUE TITLE
Error formatting vector with comment reader macro & `:keyword-respect-nl`

### ISSUE THREAD

**Issue by RolT** [kkinnear/zprint#206]
Hello, I get an error when trying to format a file with zprint.

With this source file:
```clojure
[#_a
 1]
```
And this config:
```edn
{:style [:keyword-respect-nl]}
```

I'm getting the following error: `java.lang.UnsupportedOperationException`

It looks like their is a problem when the vector starts with an "ignore" reader macro `#_`.

**Comment by kkinnear** [comment]
Thank you for noticing and reporting this issue!

Yes indeed, a problem occurs when a vector starts with `#_` and you use a style that invokes `{:vector {:option-fn-first ...}}`, which `:keyword-respect-nl` does.  It tries to turn the first thing in the vector into an s-expression, and that fails.  It currently skips over comments, but not unevaluated things.

I will fix this in `1.2.0`, but that is taking longer than I expected to release.  The only workaround is to not use `:style :keyword-respect-nl` when you have `#_` as the first element of a vector.  Which I know is obvious, but I wanted to to make sure that it was stated and not just implied.  

One way to "not use `:style :keyword-respect-nl` when `#_` appears in a vector" would be to essentially turn it off prior to the offending vector using the [File Comment API](https://cljdoc.org/d/zprint/zprint/1.1.2/doc/file-comment-api#comment-formatting-directives):
```
;!zprint {:format :next :vector {:option-fn-first nil}}
```
This will turn off the `:style :keyword-respect-nl` feature from the next top level expression.  Alternatively, you could use this
zprint comment directive prior to the expression with the `#_` as the first thing in the vector if you want newlines to be respected in vectors in that expression:
```
;!zprint {:format :next :vector {:respect-nl? true}}
```
Unfortunately, there isn't any more clever workaround that I can find for this problem.

Thank you again for noticing and reporting this issue!

**Comment by RolT** [comment]
Thanks ! No need for a workaround, I just removed the offending comment form, it didn't serve any purpose.

**Comment by kkinnear** [comment]
Thanks again for noticing and reporting this issue.   Fixed in `1.2.0`.
